### PR TITLE
Changes to the `spin new` experience

### LINKED ISSUE #1525: `spin new`: provide application name without providing template
When creating a Spin application, I usually know the name I want to give the application, but I often cannot remember the template name I'd like to use (is it `http-rust`, `rust-http`, or something else?). I really wish I could `spin new $SOME_NAME` and then be presented with a picker for the template. 

This might not be possible to add though given that `spin new` first expects a template *then* a name.

### PR DESCRIPTION
**Pull request by itowlson**
Fixes #1525.

~**WIP:** There are more changes requested in #1525, but I'm getting it under way immediately because the e2e tests use `spin new` quite a bit and I want to know where they blow up sooner rather than later.  I fixed one place but I'm sure there will be others!~  This now has everything except the proposal to generate a blank component on `spin new` with no template, on which I'd like to gather more feedback before implementing.  That can be a follow-up PR if necessary.

And you don't know how much I wanted to call this PR "The new `new` thing."

### CODE REVIEW

**Review by fibonacci1729 — APPROVED** [review]
🎉 Excellent!

File: src/commands/new.rs [hunk 1]
```diff
@@ -34,9 +35,14 @@ pub struct TemplateNewCommandCore {
 
     /// The directory in which to create the new application or component.
     /// The default is the name argument.
-    #[clap(short = 'o', long = "output")]
+    #[clap(short = 'o', long = "output", group = "location")]
     pub output_path: Option<PathBuf>,
 
+    /// The directory in which to create the new application or component.
```
Comment (fibonacci1729) [inline src/commands/new.rs:41]:
```suggestion
    /// Create the application in the current working directory.
```

... Or something (comment appears to be the same from `-o`)

Comment (itowlson) [reply src/commands/new.rs:41]: oopsie boopsie!  Thanks for the catch

**Comment by itowlson** [comment]
@melissaklein24 I unlinked #16 because this doesn't currently address that.  See https://github.com/fermyon/spin/issues/1525#issuecomment-1756104398 and https://github.com/fermyon/spin/issues/1525#issuecomment-1756138632 for more context on what fixing #16 would involve.

**Review by rylev — APPROVED** [review]
🎉

**Pull request merged**